operator/main.go

```diff
@@ -89,6 +85,22 @@ func main() {
 		namespace = watchNamespace
 	}
 
+	// Initialize ingress plugin
+	ingressPlugin := controllers.NewDefaultIngress()
+	if controllers.GetEnv(controllers.ENV_ISTIO_ENABLED, "false") == "true" {
+		setupLog.Info("Enabling Istio Ingress")
+		ingressPlugin = controllers.NewIstioIngress()
+	}
+	if controllers.GetEnv(controllers.ENV_AMBASSADOR_ENABLED, "false") == "true" {
+		setupLog.Info("Enabling Ambassador Ingress")
+		ingressPlugin = controllers.NewAmbassadorIngress()
+	}
+	if controllers.GetEnv(controllers.ENV_CONTOUR_ENABLED, "false") == "true" {
+		setupLog.Info("Enabling Contour Ingress")
+		ingressPlugin = controllers.NewContourIngress()
+	}
+	ingressPlugin.AddToScheme(scheme)
```

Comment:
Should we ensure only 1 of these is activated?

Reply:
That is probably the best option for now, right now it will just override based on priority but could put something more explicit to ensure only one is enabled.

Latter on I designed the rest of the code to support multiple being enabled at the same time in order to make migrations easier. i.e I want to move from Ambassador to Istio etc.